tools/loc/sharepoint.js

```diff
@@ -84,6 +82,53 @@ function getAuthorizedRequestOption({
   return options;
 }
 
+let nextCallAfter = 0;
```

Comment:
I don't think we should be adding the fetch-with-retry functionality to the `loc/sharepoint.js` file since there is nothing specific to sharepoint. I'd rather add this to `floodgate/js/utils.js` for now.
Please note that the complete loc and floodgate backend will be externalized as part of MWPW-122323 and MWPW-125048 anyway.

Reply:
Actually this check is directly related to the response provided by Sharepoint APIs. The fetchWithRetry checks for headers returned by Sharepoint APIs. This is not specific to Floodgate or Loc but applies to anything that will rely on Sharepoint APIs. So considering that, it makes more sense for this to be in sharpeoint.js. More in this [MS documentation](https://learn.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online#how-to-handle-throttling). That said, I do agree that this will move to AIO at some point and is being added here for now for the ease of testing.

Reply:
Thanks for the context, guess I overlooked the status headers from sharepoint. I'm ok with adding this here for the time being until we move things to I/O runtime.